### PR TITLE
adding logic to push beyond the 14.04 python 2.7.6 barrier

### PR DESCRIPTION
**Pull request by JoeNunnelley**
this installs 2.7.14 (current) within the docker contiainer.  we were previously locked to the latest releast for ubuntu 14.04 which appears to be stagnant and locked to 2.7.6

### CODE REVIEW

File: runit-python/Dockerfile [hunk 1]
```diff
@@ -4,6 +4,12 @@ MAINTAINER Socrata <sysadmin@socrata.com>
 RUN DEBIAN_FRONTEND=noninteractive apt-get -y update && \
     DEBIAN_FRONTEND=noninteractive apt-get -y install python python-dev python-pip
 
+# Update the Python installed version to latest
```
Comment (iamjohnnym) [inline runit-python/Dockerfile]: I'm concerned that upgrading this version could hurt builds upstream or do something negative.

**Comment by znep** [comment]
What is the motivation for updating?

**Comment by iamjohnnym** [comment]
The base version of python is 2.7.6, which has functions that are deprecated in relation to TLS.  This is causing issues with installing pip packages from artifactory.  Various articles suggest the steps above despite recommendations of never upgrading pip.  We could use a virtualenv but we would at least need to compile a version from source.

It'll need to be done when the base image is being created since the build will fail here: [dockerize-legacy](https://github.com/socrata/ops-tools/blob/master/jenkins/dockerize/dockerize-legacy#L62)

If we do accept going this route, I cleaned up the Dockerfile a bit: https://gist.github.com/iamjohnnym/636432817567691e4f460ee5f7141cb1

**Comment by JoeNunnelley** [comment]
If you want to make this change less global you can change it in the provisionator Dockerfile only rather than at this level.

**Comment by JoeNunnelley** [comment]
closing. @iamjohnnym is going another way with this.

**Pull request closed without merging**